### ISSUE TITLE
YDK generated model doesn't allow for DELETE() operation on valid models

### ISSUE THREAD

**Issue by siming85** [CiscoDevNet/ydk-gen#233]
I want to generate this RPC:

```
<rpc message-id="101" xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
  <edit-config>
    <target>
      <running/>
    </target>
    <config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
      <native xmlns="http://cisco.com/ns/yang/ned/ios">
        <interface>
          <GigabitEthernet>
            <name>0/2/0</name>
            <ip>
              <address xc:operation="delete"/>
            </ip>
          </GigabitEthernet>
        </interface>
      </native>
    </config>
  </edit-config>
</rpc>
```

the equivalent YDK code is:

```
        ydk_obj = ned.Native.Interface.Gigabitethernet()
        ydk_obj.name = "0/2/0"
        ydk_obj.shutdown = Empty()
        ydk_obj.description = DELETE()
        ydk_obj.ip.address = DELETE()
```

but it throws error:

```

Traceback (most recent call last):
  File "/ydk/script/script.py", line 70, in cleanup
    self.intf.build_unconfig()
  File "/ydk/cisco_shared/genie/src/genie/conf/base/decorator.py", line 19, in inner
    ret = func(*args, **kwargs)
  File "/ydk/xbu_shared/genie/conf/iosxe/yang/interface.py", line 340, in build_unconfig
    return crud_service.create(ncp, ydk_obj)
  File "/ydk/lib/python3.4/site-packages/ydk/services/crud_service.py", line 62, in create
    self._execute_crud_operation_on_provider(provider, entity, 'CREATE', False)
  File "/ydk/lib/python3.4/site-packages/ydk/services/crud_service.py", line 165, in _execute_crud_operation_on_provider
    only_config
  File "/ydk/lib/python3.4/site-packages/ydk/providers/netconf_provider.py", line 84, in encode
    return self.sp_instance.encode(entity, operation, only_config)
  File "/ydk/lib/python3.4/site-packages/ydk/providers/_provider_plugin.py", line 114, in encode
    root = self._encode_edit_request(root, entity, operation)
  File "/ydk/lib/python3.4/site-packages/ydk/providers/_provider_plugin.py", line 432, in _encode_edit_request
    self._encode_epilogue(entity, root, optype)
  File "/ydk/lib/python3.4/site-packages/ydk/providers/_provider_plugin.py", line 406, in _encode_epilogue
    XmlEncoder().encode_to_xml(entity, root, optype)
  File "/ydk/lib/python3.4/site-packages/ydk/providers/_encoder.py", line 48, in encode_to_xml
    validate_entity(entity, optype)
  File "/ydk/lib/python3.4/site-packages/ydk/providers/_validator.py", line 40, in validate_entity
    validate_entity_delegate(entity, optype, errors)
  File "/ydk/lib/python3.4/site-packages/ydk/providers/_validator.py", line 66, in validate_entity_delegate
    if hasattr(value, '_has_data') and not value._has_data():
  File "/ydk/lib/python3.4/site-packages/ydk/models/ned.py", line 1148133, in _has_data
    if self.address is not None and self.address._has_data():
AttributeError: 'DELETE' object has no attribute '_has_data'

```

changing it to:

```
        ydk_obj = ned.Native.Interface.Gigabitethernet()
        ydk_obj.name = "0/2/0"
        ydk_obj.shutdown = Empty()
        ydk_obj.description = DELETE()
        ydk_obj.ip.address.primary.address = DELETE()
        ydk_obj.ip.address.primary.mask = DELETE()
```

works... but that's a more precise RPC operation.

**Comment by psykokwak4** [comment]
Hi Siming,

In YDK [delete documentation](http://ydk.cisco.com/py/docs/ydk.types.html#ydk.types.ydk.types.DELETE), the delete object is supposed to be used in leaf level, maybe you could try CRUD update operation on `ydk_obj.ip.address`?

**Comment by siming85** [comment]
a crud update on this would mean two separate RPC instead of one.

`xc:operation="delete"` makes it a one step approach, so we should be able to do call `DELETE()` on anything.

**Comment by abhikeshav** [comment]
Hi Siming,

I agree regarding the usage of DELETE() object. The CRUD service was meant to handle simple delete operations on list/container using crud.delete operation. To allow deleting leafs, we had introduced the the ydk DELETE() object.

However for more granular netconf configurations, we had created the [netconf service](http://ydk.cisco.com/py/docs/ydk.services.html#netconfservice-provides-api-s-to-execute-netconf-operations). However, the full implementation of netconf service to support the various edit operations (delete, replace  etc) is in our backlog of TODO items.

Thanks,
Abhi
